### DISCUSSION TITLE
T-Display-S3 Trouble uploading sketches.

### DISCUSSION THREAD

**Discussion by joniejava** [Bodmer/TFT_eSPI#3056] [Compatible displays and setup files]
Hi i am quite new to the displays and hope someone can help me with possible workflow problems. Ive had a few displays and it seems every time i upload an example sketch the board loads it but the screen is just black. I have tried using the binary upload and managed to get a sketch to work and i try another sketch and it goes black. Im using Arduino IDE. Any help is greatly appreciated. I have since been getting used to including all dependant files but have class TFT_eSprite' has no member named ' errors now. Any help pls. Im using an esp32 btw.

**Comment by projectitis** [comment]
It sounds like maybe you haven't specified the correct definitions for the T-Display-S3 screen?
I'm using that board right now, and these are mine. I also had a black screen until I realised I had accidentally missed the backlight pin.

```
USER_SETUP_LOADED 1
ST7789_DRIVER 1
TFT_PARALLEL_8_BIT 1
TFT_WIDTH 170
TFT_HEIGHT 320
TFT_CS 6
TFT_DC 7
TFT_WR 8
TFT_RD 9
TFT_RST 5
TFT_D0 39
TFT_D1 40
TFT_D2 41
TFT_D3 42
TFT_D4 45
TFT_D5 46
TFT_D6 47
TFT_D7 48
TFT_BL 38
TOUCH_CS -1
SPI_FREQUENCY 27000000
```

**Reply by joniejava** [reply]
Hi Thanks for your advice. How do i implement this?

**Reply by projectitis** [reply]
I don't use Arduino IDE, but [this video I found](https://youtu.be/b8254--ibmM?t=294) might help you (from about 4:40 onwards).
He's using an older TTGO board so you'll need to select the correct one instead.
Its the "User_setup.h" instructions where you need the values above.
